### ISSUE TITLE
Incompatibility with Apache Groovy 3.0.19

### ISSUE THREAD

**Issue by puneetbehl** [grails/grails-data-mapping#1767]
Compiling the project against Apache Groovy 3.0.19 gives following error:

```shell
> Task :grails-datastore-gorm-validation:compileGroovy
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/behl/GitHub/grails-data-mapping/grails-datastore-gorm-validation/src/main/groovy/org/grails/datastore/gorm/validation/constraints/MinConstraint.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
startup failed:
/Users/behl/GitHub/grails-data-mapping/grails-datastore-gorm-validation/src/main/groovy/org/grails/datastore/gorm/validation/constraints/registry/DefaultConstraintRegistry.groovy: 38: [Static type checking] - Incompatible generic argument types. Cannot assign java.util.Map <String, java.util.List> to: java.util.Map <String, List>
 @ line 38, column 70.
   ntFactory>> factoriesByName = new Concur
                                 ^

/Users/behl/GitHub/grails-data-mapping/grails-datastore-gorm-validation/src/main/groovy/org/grails/datastore/gorm/validation/constraints/registry/DefaultConstraintRegistry.groovy: 41: [Static type checking] - Incompatible generic argument types. Cannot assign java.util.Map <Class, java.util.List> to: java.util.Map <Class, List>
 @ line 41, column 91.
   ntFactory>> factoriesByType = new Concur
                                 ^

2 errors


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':grails-datastore-gorm-validation:compileGroovy'.
> Compilation failed; see the compiler error output for details.

```

The Groovy code which is throwing this error is:

```groovy
protected Map<String, List<ConstraintFactory>> factoriesByName = new ConcurrentHashMap<String, List<ConstraintFactory>>().withDefault { String name ->
    return []
}
protected Map<Class<? extends Constraint>, List<ConstraintFactory>> factoriesByType = new ConcurrentHashMap<Class<? extends Constraint>, List<ConstraintFactory>>().withDefault { Class<? extends Constraint> type ->
    return []
}
```

**Comment by puneetbehl** [comment]
@paulk-asert Could you please help with this?

**Comment by matrei** [comment]
@puneetbehl Hmm, those lines where changed in #1754. Do you have the 8.0.x branch checked out?

**Comment by puneetbehl** [comment]
I will crosscheck next week and revert back.

**Comment by matrei** [comment]
Fixed by #1754
